### PR TITLE
lightning: improve log

### LINKED ISSUE #46687: "failed to get StoreInfo from pd http api" should be WARN instead of ERROR
## Enhancement

https://github.com/pingcap/tidb/blob/66fe053b468e8d4c1f182b98e796ca48075809a7/br/pkg/lightning/backend/local/local.go#L1356

It does not affect main logic

### LINKED ISSUE #54478: fast ddl logs are complete
## Feature Request

**Is your feature request related to a problem? Please describe:**

The placment rule is incorrectly configured, causing fast ddl to be stuck in region scatter. However, it is difficult to locate the cause in logs. It is hoped that relevant logs can be printed to locate the problem easily.

For example： ddl_worker [warn] scatter too slow

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #46687 close #54478

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/util/logutil/log.go [hunk 1]
```diff
@@ -395,3 +396,25 @@ func proxyFields() []zap.Field {
...
 	return fields
 }
+
+// SampleLoggerFactory returns a factory function that creates a sample logger.
+// the logger is used to sample the log to avoid too many logs, it will only log
+// the first 'first' log entries with the same level and message in 'tick' time.
+// NOTE: Because we need to record the log count with the same level and message
+// in this specific logger, the returned factory function will only create one logger.
+// this logger support at most 4096 types of logs with the same level and message.
+func SampleLoggerFactory(tick time.Duration, first int, fields ...zap.Field) func() *zap.Logger {
```
Comment (D3Hunter) [inline pkg/util/logutil/log.go:406]: conflicts with https://github.com/pingcap/tidb/pull/54438

File: br/pkg/utils/retry.go [hunk 2]
```diff
@@ -246,6 +247,8 @@ func WithRetryV2[T any](
 	return *new(T), allErrors // nolint:wrapcheck
 }
 
+var sampleLoggerFactory = logutil.SampleLoggerFactory(time.Minute, 3, zap.String(logutil.LogFieldCategory, "br-utils"))
```
Comment (D3Hunter) [inline br/pkg/utils/retry.go:250]:
```suggestion
var sampleLoggerFactory = logutil.SampleLoggerFactory(time.Minute, 3, zap.String(logutil.LogFieldCategory, "utils"))
```
as lightning will depends on it indirectly, maybe not appear `br`

Comment (lance6716) [reply br/pkg/utils/retry.go:250]: I also want to add something to show it's a "sample logger", so when reading log we don't think the function is really executed for 1 minutes. Do you think `zap.String(logutil.LogFieldCategory, "utils-sample-logger")` is good?

File: pkg/util/logutil/log.go (hunk 1 above)
Comment (lance6716) [reply pkg/util/logutil/log.go:406]: I hope git can automatic merge the commit with same file diff

File: br/pkg/utils/retry.go (hunk 2 above)
Comment (D3Hunter) [reply br/pkg/utils/retry.go]: maybe `zap.String("sampled", "")`?

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by D3Hunter — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**